docs/contributing.rst

```diff
@@ -335,22 +335,23 @@ Release procedure
 .. note:: 
 
    Most of the release process is now handled by github workflow which should
-   automatically push a release to PyPI if a tag is pushed. 
+   automatically push a release to PyPI if a tag is pushed.
 
-Checkout and update the main branch::
+Before releasing, make sure that all pull requests which will be
+included in the release have been properly documented in
+`docs/release.rst`.
 
-    $ git checkout main
-    $ git pull
+To make a new release, go to
+https://github.com/zarr-developers/zarr-python/releases and
+click "Draft a new release". Choose a version number prefixed
+with a `v` (e.g. `v0.0.0`) and set the description to:
```

Comment:
As noted in issue ( https://github.com/zarr-developers/zarr-python/issues/1118 ), we could alternatively use a GitHub action to automate this step of adding and linking the release notes, in which case we would only need to create a tag on GitHub to kick off the release process